### PR TITLE
Implementation of sorting

### PR DESCRIPTION
**Pull request by KacperKotlewski**
I've implemented what we discussed in [this issue](https://github.com/mxkooo/JavaScriptGame/issues/1)
I hope it will help you with your project
If you have any questions just ask

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -120,6 +120,22 @@ const gameCategories = [
   },
 ]
 
+function get_number_from_level(level)
```
Comment (KacperKotlewski) [inline script.js:123]: this function takes a level and checks if this value is in the values object,
if it returns a value for particular level
else throws error

File: script.js [hunk 2]
```diff
@@ -120,6 +120,22 @@ const gameCategories = [
...
+{
+  const values = {'easy':20,'medium':50,'hard':100}
+  if (level in values) { 
+    return values[level] 
+  }
+  alert('Error, please check console');
+  throw 'Provided level:('+typeof(level)+':"'+level+'") is not supported in values';
+}
+
+function sort_questions(questions) {
```
Comment (KacperKotlewski) [inline script.js:133]: sorting function based on [this thread](https://stackoverflow.com/questions/17684921/sort-json-object-in-javascript#:~:text=131-,First,-off%2C%20that%27s%20not)

File: script.js [hunk 3]
```diff
@@ -131,20 +147,14 @@ function giveCategory(category) {
   column.append(genreTitle)
   game.append(column)
 
+  sort_questions(category.questions)
```
Comment (KacperKotlewski) [inline script.js:150]: Implemented sorting

File: script.js [hunk 4]
```diff
@@ -131,20 +147,14 @@ function giveCategory(category) {
...
-    if (question.level == 'easy') {
-      card.innerHTML = 20
-    }
-    if (question.level == 'medium') {
-      card.innerHTML = 50
-    }
-    if (question.level == 'hard') {
-      card.innerHTML = 100
-    }
+    card.innerHTML = get_number_from_level(question.level)
```
Comment (KacperKotlewski) [inline script.js:157]: changed if chain to invoke of function

**Comment by mxkooo** [comment]
Kacper, thanks a lot for help in solving this problem. 😁

**Comment by KacperKotlewski** [comment]
NP! :D

**Comment by KacperKotlewski** [comment]
idk why you just don't marge it instead of copy paste but it's your choice, I've closed the request because you paste this into your code and the request is no more necessary but next time marge is better.

**Pull request closed without merging**